doc/migrate-to-pymongo4.rst

```diff
@@ -321,6 +321,13 @@ Can be changed to this::
 
 .. _'system.profile' collection: https://docs.mongodb.com/manual/reference/database-profiler/
 
+Database.__bool__ raises NotImplementedError
+..............................................
+:class:`~pymongo.database.Database` now raises an error upon evaluating as a
+Boolean, please use the syntax ``if database is not None:`` as opposed to
+the previous syntax which was simply ``if database:``. You must now
+explicitly compare with None.
```

Comment:
Can you use the same "Code like this::...Can be changed to this::.." pattern as the other entries in this file? :
```
Code like this::

...

Can be changed to this::

...

```

Reply:
Done. Also changed it for collection.